### PR TITLE
Add cache configure script

### PR DESCRIPTION
**Pull request by ThetaSinner**
### TODO:
- [ ] CHANGELOG(s) updated with appropriate info
- [ ] Just before pressing the merge button, ensure new entries to CHANGELOG(s) are still under the _UNRELEASED_ heading

### CODE REVIEW

File: .github/workflows/test-holonix-setup.yml [hunk 1]
```diff
@@ -35,3 +35,18 @@ jobs:
...
+          CACHE_DOMAIN_SUFFIX_OVERRIDE: .cachix.org
+          CACHE_PUBLIC_KEY_OVERRIDE: holochain-ci.cachix.org-1:5IUSkZc0aoRS53rfkvH9Kid40NpyjwCMCzwRTXy+QN8=
+        run: |
+          set -xe
+
+          bash <(curl https://raw.githubusercontent.com/holochain/holochain/${{ github.ref_name }}/configure-cache.sh) -s use holochain-ci 
+
+          nix_system=$(nix show-config --json | jq -r .system.value)
+
+          nix build -L -j0 \
```
Comment (steveej) [inline .github/workflows/test-holonix-setup.yml:50]:
```suggestion
          nix build --refresh -L -j0 \
```

File: .github/workflows/test-holonix-setup.yml [hunk 2]
```diff
@@ -35,3 +34,29 @@ jobs:
...
+              source $file > /dev/null 2>&1
+              if command -v nix; then break; fi
+          done
+          set -e
+
+          bash <(curl https://raw.githubusercontent.com/holochain/holochain/${{ github.sha }}/configure-cache.sh) use holochain-ci
+
+          nix_system=$(nix show-config --json | jq -r .system.value)
+
+          nix build -L --refresh -j0 \
```
Comment (steveej) [inline .github/workflows/test-holonix-setup.yml:60]: would be ideal if this was cached but it depends on https://github.com/holochain/holochain/pull/2816 AFAIU

```suggestion
          nix develop --build -L --refresh -j0 \
```

**Review by steveej — APPROVED** [review]
awesome work, thank you!

i've got some suggestions that would be nice to get in now or later.

File: .github/workflows/test-holonix-setup.yml [hunk 3]
```diff
@@ -80,12 +80,14 @@ jobs:
...
+          # Now actually configure the cache
           bash <(curl https://raw.githubusercontent.com/holochain/holochain/${{ github.sha }}/configure-cache.sh) use holochain-ci
-          do_nix_build
           
           # Give the daemon a chance to restart
-          sleep 1
+          sleep 5
+          
+          # and try a build where everything is expected to be cached
+          do_nix_build
```
Comment (steveej) [inline .github/workflows/test-holonix-setup.yml:90]: could we use the nix-check-cache here? if not, we should probably adapt that one to be usable here as that it's purpose.

Comment (ThetaSinner) [reply .github/workflows/test-holonix-setup.yml:90]: Yes we absolutely could use it, there's nothing in it that would struggle here, just need a way to grab the binary. I'll stick in a TODO for now

Comment (steveej) [reply .github/workflows/test-holonix-setup.yml:91]: wouldn't a separate action step work as it's already exposed as an action?

**Pull request merged**